app/_components/PoiCard.tsx

```diff
@@ -104,13 +120,64 @@ export function PoiCard({
     setCollect(true);
     setShowPopup && setShowPopup(false);
     setGuessPoiPosition &&
-    setGuessPoiPosition({
-      latitude: payload.exact_latitude,
-      longitude: payload.exact_longitude,
-    });
+      setGuessPoiPosition({
+        latitude: payload.exact_latitude,
+        longitude: payload.exact_longitude,
+      });
     payload.is_completed = true;
-  }
+  };
 
+  //handle hints
+  const handleGetHintOnClick = async (
+    user: User,
+    pin: Pin | null,
+    userCoordinates: Coordinates | null
+  ) => {
+    try {
+      if (!user) throw "Not logged in";
+      if (!pin) throw "No pin to track";
+      if (!userCoordinates) throw "No user coordinates";
+
+      await GetHints(user, payload);
+      updatePoiHint();
+    } catch (error) {
+      console.error("Error", error);
+    }
+  };
+  //update POI but for hints?
+  const updatePoiHint = () => {
+    setIsOpen(!isOpen);
+  };
+
+  //get hints
+  const GetHints = async (user: User, pin: Pin): Promise<Response | void> => {
+    try {
+      if (!user) throw "Not logged in"; //error
+      if (!pin) throw "Can not get hint";
+
+      const { poi_id } = pin;
+      const response: Response = await fetch(`${BASE_URL}/api/hints/${poi_id}`);
+      const data: {
+        content: string;
+        poi_id: number;
+        user_id: number;
+        hint_id: number;
+      }[] = (await response.json()) as {
+        content: string;
+        poi_id: number;
+        user_id: number;
+        hint_id: number;
+      }[];
+      const arrayOfContent: string[] = [];
+      for (let i = 0; i < data.length; i++) {
+        arrayOfContent.push(data[i].content);
+      }
```

Comment:
Small nitpick here, but if the order of hints is important, you could refactor this code like this.
```suggestion
      const arrayOfContent: string[] | undefined[] = new Array(data.length);
      for (let i = 0; i < data.length; i++) {
        arrayOfContent[i] = data[i].content;
      }
```

This will automatically create a new array of the same length, except at first everything will be `undefined`. By manually reassigning the value of each index inside the loop, you can guarantee the insertion order. `.push()` is usually fine, but doesn't always guarantee ordering.

Reply:
updated!